src/app.ts

```diff
@@ -13,14 +14,44 @@ interface TemperatureSummary {
   average: number
 }
 
+const temperatures: TemperatureReading[] = []
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  for (const reading of readings) {
+    temperatures.push(reading)
+  }
+  return
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const groupReadings = temperatures.reduce(
+    (tmp: any, obj: TemperatureReading) => {
+      tmp[obj.city] = [...(tmp[obj.city] || []), obj]
+      return tmp
+    },
+    [],
+  )
+  const readFiltered = groupReadings[city]
+  if (!groupReadings[city]) {
```

Comment:
Good type guard to make sure the city exists

Reply:
Thak you :+1: